### ISSUE TITLE
[TEALIUM SWIFT BUG]: Deadlock when creating multiple `Tealium` instances with 1.9.6

### ISSUE THREAD

**Issue by russellporter** [Tealium/tealium-swift#147]
**Your Environment**
iOS Version: 13.6
Xcode version: 11.6 (11E708)
Swift version: 5
Tealium Swift library version: 1.9.6
Dependency manager: CocoaPods
Dependency manager version: 1.9.1

**Describe the bug**
We have some tests of our tracking code. After upgrading to Tealium to 1.9.6, we noticed that our tests occasionally deadlock when creating an instance of `Tealium`.

**Does a workaround exist?**
Don't create more than one instance of `Tealium`.

**To Reproduce**

With the new version of the SDK, this test deadlocks, for me normally after 10-50 iterations. It works fine on 1.9.5.

```swift
import XCTest
import TealiumSwift

final class TealiumTests: XCTestCase {

    private var instances: [Tealium] = []

    private var config: TealiumConfig {
        fatalError("Add your own tealium config here")
    }

    func testTealiumDoesntDeadlockOnInit() {
        (0..<1000).forEach { i in
            print("Iteration \(i)")
            instances.append(Tealium(config: config))
        }
    }
}

```

**Expected behavior**
It doesn't deadlock.

**Comment by craigrouse** [comment]
@russellporter Thanks for reporting - we'll take a look. Is this only happening in your tests, or do you also create multiple instances of Tealium in your production code?

**Comment by russellporter** [comment]
Thanks! We only create a single instance in our production code.

**Comment by christinasund** [comment]
Hi @russellporter We just released a patch for freezes like this ([1.9.7](https://github.com/Tealium/tealium-swift/releases/tag/1.9.7)) as well as a brand new major version with many new features and improvements if you would like to give it a try: [2.0.0](https://github.com/Tealium/tealium-swift/releases/tag/2.0.0) Whichever version you choose, let us know if this solves the problem you were seeing. Thanks!

**Comment by russellporter** [comment]
I verified it's fixed with 2.0.0. Thanks for the fix!
